res/controllers/Soundless_joyMIDI_scripts.js

```diff
@@ -0,0 +1,738 @@
+            // press
+            if (!joyMIDI[group].btnShift) {
+                if (engine.getValue(group, "sync_enabled") > 0) {
+                    engine.setValue(group, "sync_enabled", 0);
+                } else {
+                    engine.setValue(group, "sync_enabled", 1);
+                    if (joyMIDI[group].syncTimerID !== null) {
+                        engine.stopTimer(joyMIDI[group].syncTimerID);
+                    }
+                    joyMIDI[group].syncLongPress = 0;
+                    joyMIDI[group].syncTimerID =
+                        engine.beginTimer(1000, "joyMIDI.syncTimerHandler('" + group + "')", true);
+                }
+            } else {
+                engine.setValue(group, "rate", 0);
+            }
+        } else {
+            // release
+            if (!joyMIDI[group].btnShift) {
+                if (joyMIDI[group].syncTimerID !== null) {
+                    if (joyMIDI[group].syncLongPress === 0 && engine.getValue(group, "sync_enabled") === 1) {
+                        engine.stopTimer(joyMIDI[group].syncTimerID);
+                        engine.setValue(group, "sync_enabled", 0);
+                    }
+                    joyMIDI[group].syncTimerID = null;
+                }
+            }
+        }
+    } else {
+        // knob enabled
+        if (value > 0) {
+            // press
+            joyMIDI[group].btnSync = PRESSED;
+        } else {
+            // release
+            if (joyMIDI[group].btnSync != ADJUSTED) {
+                if (!joyMIDI[group].btnShift) {
+                    engine.setValue(group, "beatsync", 1);
+                } else {
+                    engine.setValue(group, "rate", 0);
+                }
+            }
+            joyMIDI[group].btnSync = RELEASED;
+        }
+    }
+}
+
+joyMIDI.syncTimerHandler = function(group) {
+    joyMIDI[group].syncLongPress = 1;
+}
+
+joyMIDI.cueButton = function(channel, control, value, status, group) {
+    var sft = joyMIDI[group].btnShift;
+    var cmd = sft ? "start" : "cue_default";
+    var val = (value > 0) ? 1 : 0;
+
+    engine.setValue(group, cmd, val);
+}
+
+joyMIDI.playButton = function(channel, control, value, status, group) {
+    var sft = joyMIDI[group].btnShift;
+    var cmd = sft ? "reverse" : "play";
+
+    if (value > 0) {
+        script.toggleControl(group, cmd);
+    }
+}
+
+joyMIDI.scratchButton = function(channel, control, value, status, group) {
+    if (value > 0) {
+        joyMIDI[group].isScratchPermanent = !joyMIDI[group].isScratchPermanent;
+    }
+}
+
+joyMIDI.keyButton = function(channel, control, value, status, group) {
+    if (!joyMIDI.userSetting.knobEnable) {
+        // knob disabled
+        if (value > 0) {
+            // press
+            if (!joyMIDI[group].btnShift) {
+                engine.setValue(group, "sync_key", 1);
+            } else {
+                engine.setValue(group, "reset_key", 1);
+            }
+        } else {
+            // release
+        }
```

Comment:
No need for an else case here. You could even reduce the indentation level, by using something like this:

```js
if (value <= 0) {
    // release
    return;
}
```